tensorboard/webapp/core/store/core_types.ts

```diff
@@ -37,6 +37,8 @@ export interface CoreState {
   // For now, we want them here for Polymer interop states reasons, too.
   polymerInteropRuns: Run[];
   polymerInteropRunSelection: Set<RunId>;
+  // Number between 0 and 100.
+  sideBarWidthInPercent: number;
```

Comment:
I originally expected to see the value stored under metrics/, since it is specific to Time Series UI.  Having it under core/ makes sense when under the assumption that there will not be other resizable layouts.  Would it be better to move this field, the reducer, selector, etc to metrics/ as "(get)metricsRunSidebarWidth", if we want other features besides metrics/ to have persisted sidebar widths?  e.g. maybe the Hparams dashboard remembers its own sidebar width separately in the future.

If not, could we at least add comments to here and in the selector definition that mention it is specific to Time Series' run sidebar?

Reply:
> This change introduces new layout component that you can use to have
consistent layout in a plugin.

This component is expected to be used by other plugins as the description indicates.

For instance, I want this to be usable in npmi plugin which I have in a follow up change. I think it makes sense for all sidebar widths to be linked thus this state remains in the Core plugin.

Reply:
Thanks for clarifying, I see where this is going now.  This setting and the tb-layout component isn't to be used to make any resizable sidebar, (e.g. not to be used for cases like the Time Series settings pane), but specifically only for replacing the top-level layout for each plugin dashboard.